modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/ReadResolutionCalculator.java

```diff
@@ -224,10 +228,28 @@ public ReadResolutionCalculator(
             }
         }
 
-        // reprojection can turn a segment into a zero lenght one, in that case, fall back on
-        // the full resolution in that case
-        double minDistanceX = Math.max(minDistance, fullResolution[0]);
-        double minDistanceY = Math.max(minDistance, fullResolution[1]);
+        // reprojection can turn a segment into a zero length one
+        boolean zeroLength = Math.abs(minDistance) < DELTA; // Zero length check
+        double fullRes[] = new double[] {fullResolution[0], fullResolution[1]};
+        if (zeroLength && isReprojected) {
+            // Create a full resolution's one pixel bbox and reproject it to
+            // retrieve full resolution in read CRS
+            double x0 = requestedBBox.getMedian(0);
+            double y0 = requestedBBox.getMedian(1);
+            double x1 = x0 + fullRes[0];
+            double y1 = y0 + fullRes[1];
+            GeneralEnvelope envelope =
+                    new GeneralEnvelope(new double[] {x0, y0}, new double[] {x1, y1});
+            envelope = CRS.transform(destinationToSourceTransform, envelope);
+            GridToEnvelopeMapper mapper =
+                    new GridToEnvelopeMapper(new GridEnvelope2D(0, 0, 1, 1), envelope);
+            AffineTransform transform = mapper.createAffineTransform();
+            fullRes[0] = XAffineTransform.getScaleX0(transform);
+            fullRes[1] = XAffineTransform.getScaleY0(transform);
+        }
+        // fall back on the full resolution when zero length
+        double minDistanceX = zeroLength ? fullRes[0] : minDistance;
```

Comment:
This doens't look like a fall-back, maybe I'm missing something, in my understanding the ``if (zeroLength && isReprojected)`` is there to make sure that if the read bounding-box is reprojected or is empty (zero length) then we fallback? on creating a full resolution on pixel bounding-box.

Reply:
The previous code was simply doing a Max(fullRes, minDistance) to prevent zeroLength (minDistance = 0).
The problem was that in case of reprojection (even if not zeroLength) fullRes can be > minDistance so fullRes was taken (20 meters instead of 0.0002 degrees) returning back huge pixels.

With the new code:
- zeroLength is checked via a zero comparison: abs(minDistance) > 1E-10 .
- in case of zero length minDistance will not be used in favor of fullRes.
- in case of reprojection, fullRes need to be recomputed to report resolution in proper CRS. However this computation is only performed when zeroLength since it will only be used in that case. (minDistance will be used otherwise)

Does it make sense?

Reply:
Ok understood, thanks.